qiskit_machine_learning/datasets/ad_hoc.py

```diff
@@ -319,40 +193,31 @@ def ad_hoc_data(
         )
 
 
-def sample_ad_hoc_data(sample_total, test_size, n):
-    """returns sample ad hoc data"""
-
-    class_labels = [r"A", r"B"]  # copied from ad_hoc_data()
-    count = 0
-    if n == 2:
-        count = 100
-    elif n == 3:
-        count = 20
-
-    label_train = np.zeros(2 * test_size)
-    sample_a = [[0 for x in range(n)] for y in range(test_size)]
-    sample_b = [[0 for x in range(n)] for y in range(test_size)]
-    t_r = 0
-    while t_r < (test_size):
-        draw1 = algorithm_globals.random.choice(count)
-        draw2 = algorithm_globals.random.choice(count)
-        if sample_total[draw1][draw2] == +1:
-            sample_a[t_r] = [2 * np.pi * draw1 / count, 2 * np.pi * draw2 / count]
-            t_r += 1
-
-    t_r = 0
-    while t_r < (test_size):
-        draw1 = algorithm_globals.random.choice(count)
-        draw2 = algorithm_globals.random.choice(count)
-        if sample_total[draw1][draw2] == -1:
-            sample_b[t_r] = [2 * np.pi * draw1 / count, 2 * np.pi * draw2 / count]
-            t_r += 1
-    sample_train = [sample_a, sample_b]
-    for lindex in range(test_size):
-        label_train[lindex] = 0
-    for lindex in range(test_size):
-        label_train[test_size + lindex] = 1
-    label_train = label_train.astype(int)
-    sample_train = np.reshape(sample_train, (2 * test_size, n))
-    test_input = {key: (sample_train[label_train == k, :])[:] for k, key in enumerate(class_labels)}
-    return test_input
+def _sample_ad_hoc_data(sample_total, xvals, num_samples, n):
+    count = sample_total.shape[0]
+    sample_a, sample_b = [], []
+    for i, sample_list in enumerate([sample_a, sample_b]):
+        label = 1 if i == 0 else -1
+        while len(sample_list) < num_samples:
+            draws = tuple(algorithm_globals.random.choice(count) for i in range(n))
```

Comment:
Not for this PR, is there any plan to have a way to control the seed in this undeterministic parts?